JustLog/Classes/LogstashDestination.swift

```diff
@@ -60,31 +78,45 @@ public class LogstashDestination: BaseDestination  {
     }
 
     public func forceSend(_ completionHandler: @escaping (_ error: Error?) -> Void  = {_ in }) {
-        
-        if self.logsToShip.count == 0 || self.socketManager.isConnected() {
-            completionHandler(nil)
-            return
-        }
 
         self.completionHandler = completionHandler
-        
-        logDispatchQueue.addOperation { [weak self] in
-            self?.socketManager.send()
-        }
+
+        writeLogs()
     }
     
     func writeLogs() {
         
         self.logDispatchQueue.addOperation{ [weak self] in
             
-            guard let `self` = self else { return }
+            guard let self = self else { return }
             
-            for log in self.logsToShip.sorted(by: { $0.0 < $1.0 }) {
-                let logData = self.dataToShip(log.1)
-                self.socketManager.write(logData, withTimeout: self.socketManager.timeout, tag: log.0)
+            let logsInQueue = self.logsInSocketQueue
+            let unprocessedLogs = self.logsToShip.filter({ (tag: Int, _) -> Bool in
+                !logsInQueue.contains(where: { $0 == tag })
+            })
+            guard unprocessedLogs.count > 0 else {
```

Comment:
```suggestion
            guard !unprocessedLogs.isEmpty else {
```

Reply:
Why is this prefered, out of interest?

Reply:
`isEmpty` describes what you're expecting, rather than specifically the definition of what you're expecting.

Reply:
And depending on the data structure count could be a non constant time operation